Rollup of the these upstream ALSA commits:

0edb94543092535a2c6ef33e7285004168ca73d7
ALSA: hda - Fix probe errors on Dell Studio Desktop

ef1681d82f4bc2d9e023519f0bedb86519d10c43
ALSA: hda - Add probe_mask quirk for Medion MD96630

38f1df27e3191d76e983cb9c6b4392582fd32fda
ALSA: hda - Add probe_mask default for Toshiba laptop with ALC268

--- vanilla-2.6.27.19.orig/sound/pci/hda/hda_intel.c
+++ vanilla-2.6.27.19/sound/pci/hda/hda_intel.c
@@ -1967,6 +1967,12 @@ static struct snd_pci_quirk position_fix
 	SND_PCI_QUIRK(0x1028, 0x01cc, "Dell D820", POS_FIX_LPIB),
 	SND_PCI_QUIRK(0x1028, 0x01de, "Dell Precision 390", POS_FIX_LPIB),
 	SND_PCI_QUIRK(0x1043, 0x813d, "ASUS P5AD2", POS_FIX_LPIB),
+	/* broken BIOS */
+	SND_PCI_QUIRK(0x1028, 0x20ac, "Dell Studio Desktop", 0x01),
+	/* including bogus ALC268 in slot#2 that conflicts with ALC888 */
+	SND_PCI_QUIRK(0x17c0, 0x4085, "Medion MD96630", 0x01),
+	/* conflict of ALC268 in slot#3 (digital I/O); a temporary fix */
+	SND_PCI_QUIRK(0x1179, 0xff00, "Toshiba laptop", 0x03),
 	{}
 };
 
